include/aspect/material_model/reaction_model/tian2019_solubility.h

```diff
@@ -0,0 +1,159 @@
+/*
+  Copyright (C) 2024 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef _aspect_material_reaction_melt_tian2019_solubility_h
+#define _aspect_material_reaction_melt_tian2019_solubility_h
+
+#include <aspect/material_model/interface.h>
+#include <aspect/simulator_access.h>
+#include <aspect/melt.h>
+#include <aspect/utilities.h>
+
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    using namespace dealii;
+
+    namespace ReactionModel
+    {
+
+      /**
+       * A melt model that calculates melt fraction and entropy change
+       * according to the melting model for dry peridotite of Katz, 2003.
+       * This also includes a computation of the latent heat of melting (if the latent heat
+       * heating model is active).
+       *
+       * These functions can be used in the calculation of melting and melt transport
+       * in the melt_simple material model and can be extended to other material models
+       *
+       * @ingroup ReactionModel
+       */
+      template <int dim>
+      class Tian2019Solubility : public ::aspect::SimulatorAccess<dim>
+      {
+        public:
+          // constructor
+          Tian2019Solubility();
+
+          /**
+           * Declare the parameters this function takes through input files.
+           */
+          static
+          void
+          declare_parameters (ParameterHandler &prm);
+
+          /**
+           * Read the parameters from the parameter file.
+           */
+          void
+          parse_parameters (ParameterHandler &prm);
```

Comment:
Please put these functions at the end of the list of public functions here and in the source file. This is just a convention we (try to) follow, but it makes navigating the source files easier.
A typical class structure in ASPECT looks like this (with many functions being optional):

```

class ClassName
{
  constructor();

  initialize();

  all_other_minor_functions();

  main_function(in this case melt_fraction);

  declare_parameters();

  parse_parameters();

  private:

  ... all member variables and private functions
}
```